scripts/AgathasRoom.asm

```diff
@@ -26,18 +26,18 @@ AgathaShowOrHideExitBlock:
 	predef_jump ReplaceTileBlock
 
 ResetAgathaScript:
-	xor a
+	xor a ; SCRIPT_AGATHASROOM_DEFAULT
 	ld [wAgathasRoomCurScript], a
 	ret
 
-AgathasRoom_ScriptPointers:
-	dw AgathaScript0
-	dw DisplayEnemyTrainerTextAndStartBattle
-	dw AgathaScript2
-	dw AgathaScript3
-	dw AgathaScript4
+	def_script_pointers AgathasRoom
+	dw_const AgathasRoomDefaultScript,              SCRIPT_AGATHASROOM_DEFAULT
+	dw_const DisplayEnemyTrainerTextAndStartBattle, SCRIPT_AGATHASROOM_AGATHA_START_BATTLE
+	dw_const AgathasRoomAgathaEndBattleScript,      SCRIPT_AGATHASROOM_AGATHA_END_BATTLE
+	dw_const AgathasRoomPlayerIsMovingScript,       SCRIPT_AGATHASROOM_PLAYER_IS_MOVING
+	dw_const AgathasRoomStubScript,                 SCRIPT_AGATHASROOM_STUB
 
-AgathaScript4:
+AgathasRoomStubScript:
```

Comment:
What about `NoopScript` instead of `StubScript`? (and `SCRIPT_AGATHASROOM_NOOP`)

Reply:
I chose Stub to go along with the available options for `unused` `unreferenced` `stub` from https://github.com/pret/pokecrystal/wiki/Code-cleanup.

In this case; from what I've been able to tell.. Not only does this not do anything; but I think its never referenced. (Could be wrong.) If you feel Noop is a better name I can go with that.

Reply:
Gen 2 uses "Dummy" for this sort of script.

Reply:
(Maybe what I'm about to say is less relevant in this specific instance, if as you say this script is actually unreferenced, but in general (and for consistency)...)

"Stub" and "Dummy" sound a little more accidental whereas as "Noop" sounds deliberate, and in a lot of cases it *is* deliberate to transition form a non-trivial script to a noop script once conditions are satisfied.

So I think "Noop" makes intentions clearer in the code. Like in Museum1F, for example:
```
Museum1FText1:
	text_asm
	[...]
	[...]
	ld a, SCRIPT_MUSEUM1F_NOOP
	ld [wMuseum1FCurScript], a
	jr Museum1FScriptEnd
```
This is to disable script 0 from watching the player's coord each time they take a step.